cubedash/testutils/database.py

```diff
@@ -19,129 +20,252 @@
 from datacube.model import MetadataType
 from datacube.utils import read_documents
 from datacube.utils.documents import InvalidDocException, UnknownMetadataType
+from psycopg2 import sql
 from sqlalchemy import text
 
+logger: structlog.BoundLogger = structlog.get_logger(__name__)
+
+
 GET_DB_FROM_ENV = "get-the-db-from-the-environment-variable"
+POSTGIS_IMAGE = "postgis/postgis:16-3.5"
 
 
 @pytest.fixture(scope="session")
-def postgresql_server():
-    """
-    Provide a temporary PostgreSQL server for the test session using Docker.
+def postgresql_server(worker_id: str):
+    """Provide a temporary PostgreSQL server for the test session using Docker.
 
     If already running inside Docker, and there's an ODC database configured with
     environment variables, do nothing.
 
     :return: ODC style dictionary configuration required to connect to the server
     """
-
     # If we're running inside docker already, don't attempt to start a container!
     # Hopefully we're using the `with-test-db` script and can use *that* database.
     # I think this may be copypasta from odc-tools
     if Path("/.dockerenv").exists() and (
         "ODC_DEFAULT_DB_URL" in os.environ or "ODC_POSTGIS_DB_URL" in os.environ
     ):
+        logger.warning("Running inside Docker, not starting PostGIS container")
         yield GET_DB_FROM_ENV
     else:
         client = docker.from_env()
+        logger.info("Starting PostGIS Docker Container", image=POSTGIS_IMAGE)
+        postgres_args = [
+            # Memory settings (helpful even with tmpfs)
+            "-c",
+            "shared_buffers=256MB",
+            "-c",
+            "work_mem=4MB",
+            "-c",
+            "maintenance_work_mem=64MB",
+            # Skip WAL archiving and replication setup
+            "-c",
+            "wal_level=minimal",
+            "-c",
+            "max_wal_senders=0",
+            "-c",
+            "archive_mode=off",
+            # These are redundant with tmpfs but included for completeness
```

Comment:
There is no tmpfs support in Docker for macOS, so I assume you get a regular file system there.

Reply:
There is `tmpfs` support in Docker Desktop on macOS, it uses a linux VM to host the containers. Most things work about the same as on Linux, but there are difference in networking, like not being able to run the container in the host network namespace.

I get about 5000MB/s with tmpfs, and _only_ 2500MB/s without.